src/wp-includes/blocks.php

```diff
@@ -227,9 +227,47 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		'apiVersion'      => 'api_version',
 	);
 
+	$textdomain = isset( $metadata['textdomain'] ) ? $metadata['textdomain'] : 'default';
+
 	foreach ( $property_mappings as $key => $mapped_key ) {
 		if ( isset( $metadata[ $key ] ) ) {
-			$settings[ $mapped_key ] = $metadata[ $key ];
+			$value = $metadata[ $key ];
+			switch ( $key ) {
+				case 'title':
+				case 'description':
+					// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain
+					$settings[ $mapped_key ] = translate( $value, $textdomain );
+					break;
+				case 'keywords':
+					$settings[ $mapped_key ] = array();
+					if ( ! is_array( $value ) ) {
+						continue 2;
+					}
+
+					foreach ( $value as $keyword ) {
+						// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain
+						$settings[ $mapped_key ][] = translate( $keyword, $textdomain );
+					}
+
+					break;
+				case 'styles':
+					$settings[ $mapped_key ] = array();
+					if ( ! is_array( $value ) ) {
+						continue 2;
+					}
+
+					foreach ( $value as $style ) {
+						if ( ! empty( $style['label'] ) ) {
+							// phpcs:ignore WordPress.WP.I18n.LowLevelTranslationFunction,WordPress.WP.I18n.NonSingularStringLiteralText,WordPress.WP.I18n.NonSingularStringLiteralDomain
+							$style['label'] = translate( $style['label'], $textdomain );
```

Comment:
In WP-CLI we extract the `block.json` contents with an additional gettext context.

See https://github.com/wp-cli/i18n-command/blob/0fa24efa7460564d17fa4c6e1763e806eb1613c3/src/BlockExtractor.php#L44-L68

So we'd need to use `translate_with_gettext_context()` here, with the same contexts as used by  WP-CLI.

Maybe we need to update those, which is why a schema would really help.

Reply:
I will take care of it tomorrow, one of the last remaining items on my list. Thank you for raising it.

Yes, should we follow `theme.json` and create some sort of `block-i18n.json` file? What format should it have?

Reply:
`translate_with_gettext_context` turned out to be simpler than I expected: 99f2f13 😄

Reply:
> Yes, should we follow theme.json and create some sort of block-i18n.json file? What format should it have?

I think so, yes. The format can be discussed in an issue/PR for it, but I suppose it will look very similar to the theme one